mkosi.images/system/mkosi.conf.d/10-arch/mkosi.build.chroot

```diff
@@ -24,12 +24,36 @@ ln --symbolic . "pkg/$ID/src"
 # shellcheck source=/dev/null
 . /etc/makepkg.conf
 
+CFLAGS="$CFLAGS -O${OPTIMIZATION:-0}"
+if [[ "${OPTIMIZATION:-0}" =~ ^[0g]$ ]]; then
+    CFLAGS="$CFLAGS -Wp,-U_FORTIFY_SOURCE"
+fi
+if ((LLVM)); then
+    CFLAGS="$CFLAGS -shared-libasan -fno-sanitize=function"
```

Comment:
Could you add a comment pointing to https://github.com/systemd/systemd/issues/29972? When/if it's fixed `-fno-sanitize=function` should be removed.